src/main/java/seedu/address/logic/commands/deletecommands/DeleteInternshipCommand.java

```diff
@@ -41,25 +44,26 @@ public CommandResult execute(Model model) throws CommandException {
         if (targetCompanyIndex.getZeroBased() >= lastShownList.size()) {
             throw new CommandException(Messages.MESSAGE_INVALID_COMPANY_DISPLAYED_INDEX);
         }
-        Company companyToDelete = lastShownList.get(targetCompanyIndex.getZeroBased());
-        List<Internship> currInternships = companyToDelete.getInternshipList();
-        if (targetInternshipIndex.getZeroBased() >= currInternships.size()) {
+
+        Company companyToDeleteFrom = lastShownList.get(targetCompanyIndex.getZeroBased());
+
+        if (targetInternshipIndex.getZeroBased() >= companyToDeleteFrom.getInternshipList().size()) {
             throw new CommandException(Messages.MESSAGE_INVALID_INTERNSHIP_DISPLAYED_INDEX);
         }
-        List<Internship> updatedInternships = new ArrayList<>(currInternships);
-        updatedInternships.remove(targetInternshipIndex.getZeroBased());
-        Company updatedCompany = new Company(companyToDelete.getCompanyName(),
-                companyToDelete.getCompanyPhone(), companyToDelete.getCompanyEmail(),
-                companyToDelete.getCompanyDescription(), companyToDelete.getTags(),
-                updatedInternships);
 
-        model.setCompany(companyToDelete, updatedCompany);
-        model.updateFilteredCompanyList(Model.PREDICATE_SHOW_ALL_COMPANIES);
+        Internship internshipToDelete = companyToDeleteFrom.getInternshipAtIndex(targetInternshipIndex);
+        companyToDeleteFrom.removeInternship(internshipToDelete);
+
+        companyToDeleteFrom.updateFilteredInternshipList(PREDICATE_SHOW_ALL_INTERNSHIPS);
+        // This helps to "reset" the company list UI, otherwise the company card will not update.
+        model.updateFilteredCompanyList(PREDICATE_SHOW_NO_COMPANIES);
+        model.updateFilteredCompanyList(PREDICATE_SHOW_ALL_COMPANIES);
         return new DisplayableCommandResult(
                 String.format(MESSAGE_SUCCESS,
                         this.targetInternshipIndex.getOneBased(),
-                        companyToDelete.getCompanyName()),
-                companyToDelete);
+                        companyToDeleteFrom.getCompanyName()),
+                companyToDeleteFrom,
+                String.format(DISPLAY_MESSAGE_SUCCESS, this.targetInternshipIndex.getOneBased()));
```

Comment:
Just a nit but I think the indentation level should be the same as line 64. What do you think?